### ISSUE TITLE
Installation is allowed on node versions < 8

### ISSUE THREAD

**Issue by brookr** [sonarwhal/sonar#511]
Please tell us about your:

## Environment

* __`sonar` version:__ v0.6.3
* __`Node.js` version:__ v7.4.0
* __`npm` version:__ 5.3.0

## `sonar` configuration
Could not configure, because it errored on init.

Error:
```
 > sonar --init
/usr/local/lib/node_modules/@sonarwhal/sonar/dist/src/bin/sonar.js:27
const run = async () => {
                  ^
SyntaxError: Unexpected token (
    at Object.exports.runInThisContext (vm.js:78:16)
    at Module._compile (module.js:543:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
    at bootstrap_node.js:535:3
```

</details>

**Comment by atshaas** [comment]
Upgrading Node.js fixed it for me. 
Required is Node.js v8.x. (https://sonarwhal.com/docs/user-guide/)
Your version is 7.4.0.

**Comment by alrra** [comment]
From https://docs.npmjs.com/files/package.json#engines:

> **engines**
> ...
> Unless the user has set the engine-strict config flag, this field is advisory only will produce warnings when your package is installed as a dependency.

However, testing it locally, `npm` didn't show any warnings. 

See also: 
* https://github.com/npm/npm/issues/10024
* https://github.com/npm/npm/issues/12486

---

From https://docs.npmjs.com/misc/config#engine-strict:

> **engineStrict**
> ...
> This feature was removed in npm 3.0.0
>
> Prior to npm 3.0.0, this feature was used to treat this package as if the user had set engine-> strict. It is no longer used.

From https://docs.npmjs.com/misc/config#engine-strict

> **engine-strict**
>
>    Default: false
>    Type: Boolean
>
> If set to true, then npm will stubbornly refuse to install (or even consider installing) any package that claims to not be compatible with the current Node.js version.

Doing `npm i --engine-strict @sonarwhal/sonar` did have the intended result:

```bash
npm ERR! Darwin 16.7.0
npm ERR! argv "/Users/alrra/.nvm/versions/node/v7.10.1/bin/node" "/Users/alrra/.nvm/versions/node/v7.10.1/bin/npm" "i" "--engine-strict" "@sonarwhal/sonar"
npm ERR! node v7.10.1
npm ERR! npm  v4.2.0
npm ERR! code ENOTSUP

npm ERR! notsup Unsupported engine for @sonarwhal/sonar@0.6.3: wanted: {"node":">=8.0.0"} (current: {"node":"7.10.1","npm":"4.2.0"})
npm ERR! notsup Not compatible with your version of node/npm: @sonarwhal/sonar@0.6.3
npm ERR! notsup Not compatible with your version of node/npm: @sonarwhal/sonar@0.6.3
npm ERR! notsup Required: {"node":">=8.0.0"}
npm ERR! notsup Actual:   {"npm":"4.2.0","node":"7.10.1"}

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/alrra/.npm/_logs/2017-09-14T12_54_13_340Z-debug.log
```

**Comment by molant** [comment]
I synced with @alrra earlier this morning and it looks like the most cost-effective way is to update the guidance on the website to `npm i --engine-strict @sonarwhal/sonar`

I'll do a PR for that.

**Comment by alrra** [comment]
This was fixed in https://github.com/sonarwhal/sonarwhal.com/commit/3575585157bc625b8c90e368c03240c19819a105.

**Comment by alrra** [comment]
Fixed also the documentation instructions: https://github.com/sonarwhal/sonar/commit/b2e30d6d1fd0f0a3f00a80fd04420d662ca6747a.
